echoprometheus/README.md

```diff
@@ -0,0 +1,154 @@
+# Usage
+
+```
+package main
+
+import (
+	"github.com/labstack/echo/v4"
+	"github.com/labstack/echo-contrib/prometheus/echoprometheus"
+)
+
+func main() {
+    e := echo.New()
+    // Enable metrics middleware
+    e.Use(echoprometheus.NewMiddleware("myapp"))
+    e.GET("/metrics", echoprometheus.NewHandler())
+
+    e.Logger.Fatal(e.Start(":1323"))
+}
+```
+
+
+# How to migrate
+
+## Creating and adding middleware to the application
+
+Older `prometheus` middleware
+```go
+    e := echo.New()
+    p := prometheus.NewPrometheus("echo", nil)
+    p.Use(e)
+```
+
+Never `echoprometheus` middleware
+```go
+    e := echo.New()
+    e.Use(echoprometheus.NewMiddleware("myapp")) // register middleware to gather metrics from requests
+    e.GET("/metrics", echoprometheus.NewHandler()) // register route to serve gathered metrics in Prometheus format
+```
+
+## Replacement for `Prometheus.MetricsList` field, `NewMetric(m *Metric, subsystem string)` function and `prometheus.Metric` struct
+
+`NewMetric` function allows you to create custom metrics to be registed with default Prometheus registry. Never middleware does
+not provide any abstractions for metrics and guides you to use native Prometheus metrics and register them by yourself.
+
+This can be done newer middleware as follows:
+```go
+	e := echo.New()
+
+	customRegistry := prometheus.NewRegistry() // create custom registry for your custom metrics
+	customCounter := prometheus.NewCounter( // create new counter metric. This is replacement for `prometheus.Metric` struct
+		prometheus.CounterOpts{
+			Name: "custom_requests_total",
+			Help: "How many HTTP requests processed, partitioned by status code and HTTP method.",
+		},
+	)
+	if err := customRegistry.Register(customCounter); err != nil { // register your new counter metric with metrics registry
+		log.Fatal(err)
+	}
+
+	e.Use(NewMiddlewareWithConfig(MiddlewareConfig{
+		AfterNext: func(c echo.Context, err error) {
+			customCounter.Inc() // use our custom metric in middleware. after every request increment the counter
+		},
+		Registerer: customRegistry, // use our custom registry instead of default Prometheus registry
+	}))
+	e.GET("/metrics", NewHandlerWithConfig(HandlerConfig{Gatherer: customRegistry})) // register route for getting gathered metrics data from our custom Registry
+```
+
+## Replacement for `Prometheus.MetricsPath`
+
+`MetricsPath` was used to skip metrics own route from Prometheus metrics. Skipping is no longer done and requests to Prometheus
+route will be included in gathered metrics.
+
+This feature can be restored like that:
```

Comment:
Suggestion:
To restore the old behaviour the `/metrics` path needs to be excluded from counting using the `Skipper` function: